### PR TITLE
C++: Support custom defs in SimpleRangeAnalysis

### PR DESCRIPTION
**Pull request by jbj**
@lcartey, this is based on your initial prototype, but it's a lot more complicated now. I've added extra overrides and parameters to allow specifying which variable in the `RangeSsaDefinition` is being customised.

@lcartey, can you validate that this extensibility is sufficient and practical for your use case?

### CODE REVIEW

File: cpp/ql/test/experimental/library-tests/rangeanalysis/extensibility/extensibility.ql [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import semmle.code.cpp.rangeanalysis.SimpleRangeAnalysis
+import semmle.code.cpp.rangeanalysis.RangeAnalysisUtils // for typeLowerBound
```
Comment (MathiasVP) [inline cpp/ql/test/experimental/library-tests/rangeanalysis/extensibility/extensibility.ql:2]: We don't usually place comments on imports stating why we're importing them. Is there a particular reason we're doing this here?

Comment (jbj) [reply cpp/ql/test/experimental/library-tests/rangeanalysis/extensibility/extensibility.ql]: Not really. Removed.

**Review by MathiasVP — COMMENTED** [review]
LGTM (assuming @lcartey is happy with it)

**Review by lcartey — APPROVED** [review]
Looks good to me.

**Review by geoffw0 — APPROVED** [review]
Two positive reviews and LGTM also.

**Pull request merged**